CollAction/Frontend/app/donation/DebitDetailsBox.tsx

```diff
@@ -0,0 +1,128 @@
+
+interface IDebitDetailsBoxProps {
+    stripe: any;
+    amount: number;
+    userEmail: string;
+    userName: string;
+    isRecurring: boolean;
+}
+
+interface IDebitDetailsBoxState {
+    showError: boolean;
+    error: string;
+}
+
+export default class DebitDetailsBox extends React.Component<IDebitDetailsBoxProps, IDebitDetailsBoxState> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showError: false,
+            error: ""
+        };
+    }
+
+    submitPayment(event: React.FormEvent<HTMLFormElement>) : Promise<void> {
+        event.preventDefault();
+        this.setState({ showError: false });
+
+        if (this.props.isRecurring) {
+            return this.submitPaymentSepa();
+        }
+        else {
+            return this.submitPaymentIdeal();
+        }
+    }
+
+    async submitPaymentSepa(): Promise<void> {
+        let sourceData = {
+            type: 'sepa_debit',
+            currency: 'eur',
+            owner: {
+                name: this.props.userName,
+                email: this.props.userEmail
+            },
+            mandate: {
+                notification_method: 'email' // Automatically send a mandate notification email to your customer once the source is charged
+            }
+        };
+
+        let response = await this.props.stripe.createSource(sourceData);
+        if (response.error) {
+            console.log("Unable to start SEPA Direct: " + response.error);
+            this.setState({ showError: true, error: "Unable to start SEPA Direct" })
+            return;
+        }
+
+        let initializeUrl = `/Donation/InitializeSepaDirect?sourceId=${response.source.id}&name=${encodeURIComponent(this.props.userName)}&email=${encodeURIComponent(this.props.userEmail)}&amount=${this.props.amount}`;
+        let initializeResponse = await fetch(initializeUrl, { method: 'POST' });
+        if (initializeResponse.status == 200) {
+            window.location.href = "/Donation/ThankYou";
+        } else {
+            console.log("Unable to start SEPA Direct: " + await initializeResponse.text());
+            this.setState({ showError: true, error: "Unable to start SEPA Direct" })
+        }
+    }
+
+    async submitPaymentIdeal(): Promise<void> {
+        let sourceData = {
+            type: "ideal",
+            amount: this.props.amount * 100,
+            currency: "eur",
+            statement_descriptor: "Donation CollAction",
+            owner: {
+                name: this.props.userName,
+                email: this.props.userEmail
+            },
+            redirect: {
+                return_url: window.location.origin + "/Donation/Return"
+            }
+        };
+
+        let response = await this.props.stripe.createSource(sourceData);
+        if (response.error) {
+            console.log("Unable to start iDeal: " + response.error);
+            this.setState({ showError: true, error: "Unable to start iDeal" })
```

Comment:
Since this message is shown to the user, maybe a more user-oriented text would be better?

Reply:
This only happens when there's a failure at stripe, or we have a bug. If you have suggestions..

Reply:
I added more user-friendly messages, feel free to improve them if you have ideas.